R/bayesiannetworkanalysis.R

```diff
@@ -0,0 +1,1143 @@
+  if (is.null(overTitles))
+    overTitles <- gettext("Network")
+  
+}
+
+.bayesianNetworkAnalysisPlotContainer <- function(mainContainer, network, options) {
+  
+  plotContainer <- mainContainer[["plotContainer"]]
+  
+  if (is.null(plotContainer)) {
+    plotContainer <- createJaspContainer(position = 5, dependencies = c("labelAbbreviation", "labelAbbreviationLength",
+                                                                        "legend", "variableNamesShown"))
+    mainContainer[["plotContainer"]] <- plotContainer
+  }
+  
+  .networkAnalysisNetworkPlot                    (plotContainer, network, options, method = "Bayesian")
+  .bayesianNetworkAnalysisEvidencePlot           (plotContainer, network, options)
+  .bayesianNetworkAnalysisPosteriorStructurePlot (plotContainer, network, options)
+  .bayesianNetworkAnalysisCentralityPlot         (plotContainer, network, options)
+  .bayesianNetworkAnalysisPosteriorComplexityPlot(plotContainer, network, options)
+}
+
+.bayesianNetworkAnalysisPosteriorStructurePlot <- function(plotContainer, network, options) {
+  
+  if (!is.null(plotContainer[["posteriorStructurePlotContainer"]]) || !options[["posteriorStructurePlot"]])
+    return()
+  
+  allNetworks <- network[["network"]]
+  nGraphs <- length(allNetworks)
+  
+  title <- if (nGraphs == 1L) "" else gettext("Posterior Probability Structure Plots")
+  
+  posteriorStructurePlotContainer <- createJaspContainer(title = title, position = 51, dependencies = c("posteriorStructurePlot"))
+  plotContainer[["posteriorStructurePlotContainer"]] <- posteriorStructurePlotContainer
+  
+  if (is.null(network[["network"]]) || plotContainer$getError()) {
+    posteriorStructurePlotContainer[["dummyPlot"]] <- createJaspPlot(title = gettext("Posterior Probability Structure Plot"))
+    return()
+  }
+  
+  for (v in names(allNetworks))
+    posteriorStructurePlotContainer[[v]] <- createJaspPlot(title = v)
+  
+  jaspBase::.suppressGrDevice({
+    
+    for (v in names(allNetworks)) {
+      
+      networkToPlot <- allNetworks[[v]]
+      
+      sortedStructureProbability <- as.data.frame(sort(networkToPlot$graphWeights/sum(networkToPlot$graphWeights), decreasing = TRUE))
+      colnames(sortedStructureProbability) <- "posteriorProbability"
+      plot <- ggplot2::ggplot(sortedStructureProbability, ggplot2::aes(x = 1:nrow(sortedStructureProbability), y = posteriorProbability)) +
+        ggplot2::geom_point() +
+        ggplot2::ylab("Posterior Structure Probability") +
+        ggplot2::xlab("Structure Index")  +
+        ggplot2::theme_minimal() +
+        ggplot2::theme(legend.position = c(.85, 0.25), axis.text = ggplot2::element_text(size = 20),
+              legend.background = ggplot2::element_rect(fill = NULL), panel.border = ggplot2::element_blank(),
+              axis.line = ggplot2::element_line(colour = "black", size = 1.1), axis.ticks.length = grid::unit(.2, "cm"), #Is unit from ggplot2 or grid?
```

Comment:
```suggestion
              axis.line = ggplot2::element_line(colour = "black", size = 1.1), axis.ticks.length = grid::unit(.2, "cm"),
```
`unit` is from `grid` but `ggplot2` re-exports it so it doesn't matter which you use :)